homeassistant/components/ifttt/alarm_control_panel.py

```diff
@@ -160,7 +159,7 @@ def code_format(self) -> CodeFormat | None:
         """Return one or more digits/characters."""
         if self._code is None:
             return None
-        if isinstance(self._code, str) and re.search("^\\d+$", self._code):
+        if isinstance(self._code, str) and self._code.isdigit():
```

Comment:
While the `isdigit` changes here for practical purposes do not change behavior of the code, I'm not sure if the intent in these would actually have been to match the ASCII digits `[0-9]` only, and no other Unicode ones. If that's the case, could revert back to regexps, and explicit `[0-9]` only.

I suspect there are a bunch of other cases that are inadvertently matching such non-ASCII Unicode digits across the codebase though.